src/phoenix/server/api/types/Dataset.py

```diff
@@ -221,18 +221,21 @@ async def experiments(
             before=before if isinstance(before, CursorString) else None,
         )
         dataset_id = self.id_attr
-        query = select(models.Experiment).where(models.Experiment.dataset_id == dataset_id)
-
+        query = (
+            select(
+                models.Experiment,
+                func.row_number().over(order_by=models.Experiment.id).label("ordinality"),
+            )
+            .where(models.Experiment.dataset_id == dataset_id)
+            .order_by(models.Experiment.id.desc())
+        )
         async with info.context.db() as session:
             experiments = [
-                Experiment(
-                    id_attr=experiment.id,
-                    description=experiment.description,
-                    created_at=experiment.created_at,
-                    updated_at=experiment.updated_at,
-                    metadata=experiment.metadata_,
+                to_gql_experiment(experiment, ordinality)
+                async for experiment, ordinality in cast(
+                    AsyncIterable[Tuple[models.Experiment, int]],
+                    await session.stream(query),
                 )
-                async for experiment in await session.stream_scalars(query)
             ]
```

Comment:
This cast is unnecessary when I pull your branch.

Reply:
it's not necessary because it's typed as `Any`, so casting is more explicit for downstream purposes